In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


df=pd.read_csv('StudentsPerformance.csv')
df.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33

In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   gender                       1000 non-null   object
 1   race/ethnicity               1000 non-null   object
 2   parental level of education  1000 non-null   object
 3   lunch                        1000 non-null   object
 4   test preparation course      1000 non-null   object
 5   math score                   1000 non-null   int64 
 6   reading score                1000 non-null   int64 
 7   writing score                1000 non-null   int64 
dtypes: int64(3), object(5)
memory usage: 62.6+ KB


In [3]:
df['gender'].value_counts()

female    518
male      482
Name: gender, dtype: int64

Number of male students who took   :482
Number of female students who took :518

In [4]:
df['test preparation course'].value_counts()

none         642
completed    358
Name: test preparation course, dtype: int64

In [5]:
df['parental level of education'].value_counts(normalize=True)*100

some college          22.6
associate's degree    22.2
high school           19.6
some high school      17.9
bachelor's degree     11.8
master's degree        5.9
Name: parental level of education, dtype: float64

The majority of parents have a degree from some college or an associate degreee but only 5.9% of parents have a masters degree.

In [6]:
df.drop(columns=['race/ethnicity','parental level of education','lunch'])

gender test preparation course  math score  reading score  writing score
0    female                    none           0             17             10
1    female                    none           8             24             23
2    female                    none          18             32             28
3    female                    none          11             38             32
4    female                    none          22             39             33
..      ...                     ...         ...            ...            ...
995    male               completed          99             87             81
996    male               completed         100             96             86
997    male               completed         100             97             99
998    male               completed         100            100             93
999    male               completed         100            100            100

[1000 rows x 5 columns]

In [7]:
df_females=df[df['gender']=='female']
df_females.mean()

C:\Users\Dhananjay Raja\AppData\Local\Temp\ipykernel_5204\284320075.py:2: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  df_females.mean()


math score       63.376448
reading score    72.590734
writing score    72.467181
dtype: float64

In [8]:
df_males=df[df['gender']=='male']
df_males.mean()

C:\Users\Dhananjay Raja\AppData\Local\Temp\ipykernel_5204\2942455282.py:2: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  df_males.mean()


math score       68.821577
reading score    65.545643
writing score    63.446058
dtype: float64

Based on gender Male students have a higher average in Maths but score lower in reading and writing.

In [9]:
df_prepared=df[df['test preparation course']=='completed']
df_prepared.mean()


C:\Users\Dhananjay Raja\AppData\Local\Temp\ipykernel_5204\2152815136.py:2: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  df_prepared.mean()


math score       69.966480
reading score    74.175978
writing score    74.684358
dtype: float64

In [10]:
df_notprepared=df[df['test preparation course']=='none']
df_notprepared.mean()

C:\Users\Dhananjay Raja\AppData\Local\Temp\ipykernel_5204\1138845424.py:2: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  df_notprepared.mean()


math score       63.789720
reading score    66.417445
writing score    64.457944
dtype: float64

Based on test preparation course : Students who undertook the course and completed it have higher average scores on all the subjects which means that the test preparation course was effective.

In [11]:
df_females.std()

C:\Users\Dhananjay Raja\AppData\Local\Temp\ipykernel_5204\688096923.py:1: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  df_females.std()


math score       16.029928
reading score    14.411018
writing score    14.844842
dtype: float64

In [12]:
df_males.std()

C:\Users\Dhananjay Raja\AppData\Local\Temp\ipykernel_5204\27651183.py:1: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  df_males.std()


math score       14.556411
reading score    14.149594
writing score    14.227225
dtype: float64

Female students show a greater varience in terms of marks in maths than male students which is noticeable.

In [13]:
df_prepared.std()

C:\Users\Dhananjay Raja\AppData\Local\Temp\ipykernel_5204\1494904438.py:1: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  df_prepared.std()


math score       14.521847
reading score    13.537572
writing score    13.236412
dtype: float64

In [14]:
df_notprepared.std()

C:\Users\Dhananjay Raja\AppData\Local\Temp\ipykernel_5204\2049786217.py:1: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  df_notprepared.std()


math score       15.705689
reading score    14.608896
writing score    15.041667
dtype: float64

Students who have prepared for the course and completed the preparatory course show lower variation in marks.

In [15]:
DF_Top25=df['math score'].quantile(0.75)
print(DF_Top25)

77.0


Students who have a scored marks above 77 should be given bonus marks .

In [16]:
df_top25=df[df['math score']>77]
df_top25

gender race/ethnicity parental level of education         lunch  \
430  female        group A                some college      standard   
431  female        group B           bachelor's degree  free/reduced   
432  female        group D           bachelor's degree      standard   
433  female        group D                 high school      standard   
434  female        group D           bachelor's degree  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
430               completed          78             87             91  
431                    none          78             79             76  
432                    none          78             82             79  
433                    none          78             81             80  
434                    none          78             90             93  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[231 rows x 8 columns]